lms/djangoapps/verify_student/services.py

```diff
@@ -177,22 +178,22 @@ def user_status(cls, user):
             'verification_expiry': '',
         }
 
-        # We need to check the user's most recent attempt.
-        try:
-            photo_id_verifications = SoftwareSecurePhotoVerification.objects.filter(user=user).order_by('-updated_at')
-            sso_id_verifications = SSOVerification.objects.filter(user=user).order_by('-updated_at')
-            manual_id_verifications = ManualVerification.objects.filter(user=user).order_by('-updated_at')
-
-            attempt = most_recent_verification(
-                photo_id_verifications,
-                sso_id_verifications,
-                manual_id_verifications,
-                'updated_at'
-            )
-
-        except IndexError:
-            # The user has no verification attempts, return the default set of data.
-            return user_status
+        attempt = None
+
+        verifications = active_verifications(
+            cls.verifications_for_user(user),
+            timezone.now(),
+        )
+
+        if verifications:
+            attempt = verifications[0]
+            for verification in verifications:
+                if verification.status == 'approved':
+                    # Always select the LATEST non-expired approved verification if there is such
```

Comment:
Correct me if I'm wrong but won't all (non-expired) SoftwareSecurePhotoVerification's be evaluated before SSOVerifications here? And as a result potentially not produce the 'latest' result?

Reply:
The logic in the line below this says, if the verification is `approved`, and the `created_at` date is later, choose that one instead. That logic basically select the latest approved attempt regardless of type. I also have a test case covering the case you are questioning.

Reply:
oh yup 👍